src/modules/bootloader/main.py

```diff
@@ -220,24 +222,25 @@ def install_grub(efi_directory, fw_type):
                                "--force"])
 
         # VFAT is weird, see issue CAL-385
-        efi_directory_firmware = os.path.join(efi_directory, "EFI")
-        if os.path.exists(efi_directory_firmware):
-            efi_directory_firmware = vfat_correct_case(efi_directory, "EFI")
+        install_efi_directory_firmware = vfat_correct_case(install_efi_directory, "EFI")
+        if not os.path.exists(install_efi_directory_firmware):
+            subprocess.call(["mkdir", "-p", install_efi_directory_firmware])
 
-        efi_boot_directory = os.path.join(efi_directory_firmware, "boot")
-        if os.path.exists(efi_boot_directory):
-            efi_boot_directory = vfat_correct_case(efi_directory_firmware, "boot")
-        else:
-            check_target_env_call(["mkdir", "-p", efi_boot_directory])
+        # there might be several values for the boot directory
+        # most usual they are boot, Boot, BOOT
+
+        install_efi_boot_directory = vfat_correct_case(install_efi_directory_firmware, "boot")
+        if not os.path.exists(install_efi_boot_directory):
+            subprocess.call(["mkdir", "-p", install_efi_boot_directory])
 
         # Workaround for some UEFI firmwares
-        efi_file_source = {"32": os.path.join(efi_directory_firmware, efi_bootloader_id, "grubia32.efi"),
-                           "64": os.path.join(efi_directory_firmware, efi_bootloader_id, "grubx64.efi")}
-        efi_file_target = {"32": os.path.join(efi_boot_directory, "bootia32.efi"),
-                           "64": os.path.join(efi_boot_directory, "bootx64.efi")}
-        check_target_env_call(["cp",
-                               efi_file_source[efi_bitness],
-                               efi_file_target[efi_bitness]])
+        efi_file_source = {"32": os.path.join(install_efi_directory_firmware, efi_bootloader_id, "grubia32.efi"),
+                           "64": os.path.join(install_efi_directory_firmware, efi_bootloader_id, "grubx64.efi")}
+        efi_file_target = {"32": os.path.join(install_efi_boot_directory, "bootia32.efi"),
+                           "64": os.path.join(install_efi_boot_directory, "bootx64.efi")}
+        subprocess.call(["cp",
```

Comment:
Thought off, but i see code in calamares that doing it the same way - it's was easier and less changes.